src/dparse/lexer.d

```diff
@@ -2531,3 +2594,31 @@ void main() {
     checkInvalidTrailingString(getTokensForParser(`x = q"foo`, cf, &ca));
     checkInvalidTrailingString(getTokensForParser("x = '", cf, &ca));
 }
+
+unittest
+{
+    void checkExactlyOneToken(string srcAndDst)
+    {
+        import std.conv;
+
+        LexerConfig cf;
+        StringCache ca = StringCache(16);
+
+        auto l = DLexer(srcAndDst, cf, &ca);
+        assert(l.front().type == tok!"istringLiteral", str(l.front.type));
+        assert(l.front().text == srcAndDst, l.front.text);
+        l.popFront();
+        assert(l.messages.empty);
+        assert(l.empty);
+    }
+
+    checkExactlyOneToken(`i"hello"`);
```

Comment:
looks like your tests don't include iq{} and i\`\` so might want to add this (pushed up to the dmd thing just a couple days ago. i think that is all the string forms i intend to add at this time, though i imagine eventually someone will ask for the other delimited forms too lol)

Reply:
I also added c/w/d literal suffix, I saw that was dismissed in the PR?

Reply:
Yeah, it lexes and parses them, but then issues a semantic error.